stetho/src/main/java/com/facebook/stetho/server/http/HandlerRegistry.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.server.http;
+
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+
+public class HandlerRegistry {
+  private final ArrayList<PathMatcher> mPathMatchers = new ArrayList<>();
+  private final ArrayList<HttpHandler> mHttpHandlers = new ArrayList<>();
+
+  public synchronized void register(PathMatcher path, HttpHandler handler) {
+    mPathMatchers.add(path);
+    mHttpHandlers.add(handler);
+  }
+
+  public synchronized void unregister(PathMatcher path, HttpHandler handler) {
```

Comment:
Prefer taking just a PathMatcher, finding it by index, and removing the entries at that index from both ArrayLists.

As it is, if you had something like:
path0: handler0
path1: handler1
path2: handler0

and you called:
unregister(path2, handler0)

you would end up with:
path0: handler1
path1: handler0